src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtTypesAndCodesService.java

```diff
@@ -0,0 +1,144 @@
+        this.courtDxCodesRepository = courtDxCodesRepository;
+        this.rolesProvider = rolesProvider;
+    }
+
+    public List<CourtType> getAllCourtTypes() {
+        return  courtTypeRepository.findAll()
+            .stream()
+            .map(CourtType::new)
+            .collect(toList());
+    }
+
+    public CourtTypesAndCodes getCourtTypesAndCodes(final String slug) {
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+
+        return new CourtTypesAndCodes(getCourtCourtTypes(courtEntity), courtEntity.getGbs(), getCourtDxCodes(courtEntity));
+    }
+
+    private List<CourtType> getCourtCourtTypes(final Court courtEntity) {
+
+        final List<CourtType> returnCourtTypes = courtEntity.getCourtTypes()
+                .stream()
+                .map(CourtType::new)
+                .collect(toList());
+        return mapCourtCode.mapCourtCodesForCourtTypeModel(returnCourtTypes, courtEntity);
+    }
+
+    public List<DxCode> getCourtDxCodes(final Court courtEntity) {
+        return  courtDxCodesRepository.findByCourtId(courtEntity.getId())
+            .stream()
+            .map(CourtDxCode::getDxCode)
+            .map(DxCode::new)
+            .collect(toList());
+    }
+
+    @Transactional()
+    public CourtTypesAndCodes updateCourtTypesAndCodes(final String slug, final CourtTypesAndCodes courtTypesAndCodes) {
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+        return saveNewCourtTypesAndCodes(courtEntity, courtTypesAndCodes);
+
+    }
+
+    protected CourtTypesAndCodes saveNewCourtTypesAndCodes(final Court courtEntity, final CourtTypesAndCodes courtTypesAndCode) {
+
+        final List<uk.gov.hmcts.dts.fact.entity.CourtType> courtTypeEntities = getNewCourtCourtTypesEntity(courtTypesAndCode.getCourtTypes());
+
+        //save CourtTypesAndGbsCodes
+        final Court amendedCourtEntity = saveCourtTypesAndGbsCodes(courtEntity,courtTypesAndCode,courtTypeEntities);
+
+        //save court dx codes if request is by SuperAdmin
+        if (rolesProvider.getRoles().contains(FACT_SUPER_ADMIN)) {
+            saveCourtDxCodes(amendedCourtEntity, getNewDxCodeEntity(courtTypesAndCode.getDxCodes()));
+        }
+
+        return new CourtTypesAndCodes(getCourtCourtTypes(amendedCourtEntity), amendedCourtEntity.getGbs(), getCourtDxCodes(amendedCourtEntity));
+    }
+
+
+    protected Court saveCourtTypesAndGbsCodes(final Court courtEntity, final CourtTypesAndCodes courtTypesAndCode, final List<uk.gov.hmcts.dts.fact.entity.CourtType> courtTypeEntity) {
+        if (courtEntity.getCourtTypes() == null) {
+            courtEntity.setCourtTypes(courtTypeEntity);
+        } else {
+            courtEntity.getCourtTypes().clear();
+            courtEntity.getCourtTypes().addAll(courtTypeEntity);
+        }
+
+        //remove existing code entries for court.
```

Comment:
Can we add a why comment here? Not too sure why we are setting these to be null; will they be overwritten by future changes?

Reply:
update comment with reason as the codes will be set by mapCourtCode utility